platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+
+- `utilities`
+- `monitoring`
+- `site-wide`
+- `user`
+- `static-data`
+
+These aren't _strictly_ part of the forms library, but since the forms library makes use of its parent platform code, it becomes more difficult to place clear boundaries around the forms library.
+
+### High Level Design
+
+The forms library will be consolidated into a single place and bundled as a Node package.
+This will allow VFS teams to import it as if it were any other Node module.
+
+This change should not have a noticeable functional impact on individual app teams.
+Towards the end the only change that should be made on the app side of things is changing where forms library code/styles are imported from.
+
+<!-- The source for this diagram is the `*.odg` file in the same directory, a LibreOffice Draw doc-->
+![Forms library bundling](images/forms-library-package/forms-library-package.svg)
+
+## Specifics
+
+### Detailed Design
+
+A few things will have to happen in order pave the way for additional changes that will bundle the forms library.
+
+#### Preparation
+
+- The app-specific code mentioned in [Application Entanglement](#application-entanglement) will have to be moved so that it no longer lives in the forms library.
+- Use the `no-unresolved-modules` ESLint plugin to [find & remove unused code](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8763) in the `forms` and `forms-library` directories
+
+#### Beginning the bundling
+
+- Copy what remains from `forms` into `forms-system`
+- Create `platform/packages/forms-library` and move the contents of `platform/forms-system` into it
+- Add a `package.json` file to the new `forms-library` package directory
+- [Add an alias to the `.babelrc` file](https://github.com/department-of-veterans-affairs/vets-website/blob/055d96c54e1df54138b9efc589b98e55962333b3/.babelrc#L50-L55) that redirects imports from `platform/forms` and `platform/forms-system` to the new package
+- Edit the webpack config & use a custom `sass-loader` importer to rewrite imports from `forms` to `forms-system`
+- Fix broken relative imports
+- Add babel configuration to the new package because according to [Babel's docs](https://babeljs.io/docs/en/config-files):
+
+> Searching [for a config] will stop once a directory containing a `package.json` is found, so a relative config only applies within a single package
+
+#### Clarifying boundaries
+
+The [Platform Entanglement](#platform-entanglement) section mentions that forms library code imports from "parent" platform code. This makes the boundaries less clear, so an effort will be made to reduce or eliminate the forms library importing from platform code.
+
+- Move the `static-data/labels` file into the forms library, since it is only used in connection with forms library configuration
+- Prefer code that exists in the forms library, instead of in the parent platform directory.
```

Comment:
- What does "prefer" mean? What's the action we'll take to establish a boundary?
- What if we preferred code in the platform directory? Pros? Cons?